internal/provider/aws/provider_test.go

```diff
@@ -0,0 +1,4906 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/smithy-go"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	gomock "go.uber.org/mock/gomock"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/quortex/kubestatic/api/v1alpha1"
+	"github.com/quortex/kubestatic/internal/provider"
+	"github.com/quortex/kubestatic/internal/provider/aws/mocks"
+)
+
+var _ = Describe("AWSProvider", func() {
+	var (
+		mockCtrl      *gomock.Controller
+		mockec2Client *mocks.Mockec2Client
+		provider      provider.Provider
+		clusterID     string
+	)
+
+	ctx := context.Background()
+	log := logf.Log.WithName("aws-provider-test")
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockec2Client = mocks.NewMockec2Client(mockCtrl)
+		clusterID = "cluster-id"
+
```

Comment:
Maybe it is not needed to reset clustedID at each It() ? 

```suggestion
		clusterID     = "cluster-id"
	)

	ctx := context.Background()
	log := logf.Log.WithName("aws-provider-test")

	BeforeEach(func() {
		mockCtrl = gomock.NewController(GinkgoT())
		mockec2Client = mocks.NewMockec2Client(mockCtrl)
```